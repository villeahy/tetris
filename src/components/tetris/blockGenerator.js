export default function(number) {
  switch (number) {
    case 0:
      return [{ i: 0, y: 4 }, { i: -1, y: 4 }, { i: 0, y: 5 }, { i: -1, y: 5 }]
    case 1:
      return [{ i: -2, y: 4 }, { i: -1, y: 4 }, { i: 0, y: 4 }, { i: 0, y: 5 }]
    case 2:
      return [{ i: -2, y: 5 }, { i: -1, y: 5 }, { i: 0, y: 4 }, { i: 0, y: 5 }]
    case 3:
      return [{ i: -1, y: 4 }, { i: -1, y: 5 }, { i: 0, y: 5 }, { i: 0, y: 6 }]
    case 4:
      return [{ i: -1, y: 5 }, { i: -1, y: 6 }, { i: 0, y: 4 }, { i: 0, y: 5 }]
    case 5:
      return [{ i: -3, y: 4 }, { i: -2, y: 4 }, { i: -1, y: 4 }, { i: 0, y: 4 }]
    case 6:
      return [{ i: -2, y: 4 }, { i: -1, y: 4 }, { i: -1, y: 5 }, { i: 0, y: 4 }]
    default:
      return [{ i: 0, y: 4 }, { i: -1, y: 4 }, { i: 0, y: 5 }, { i: -1, y: 5 }]
  }
}
