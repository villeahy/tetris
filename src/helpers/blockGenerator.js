export function blockGenerator(number) {
  switch (number) {
    case 0:
      return {
        value: 1,
        turn: 0,
        coords: [
          { row: 0, column: 4 },
          { row: -1, column: 4 },
          { row: 0, column: 5 },
          { row: -1, column: 5 }
        ]
      };
    case 1:
      return {
        value: 2,
        turn: 0,
        coords: [
          { row: -2, column: 4 },
          { row: -1, column: 4 },
          { row: 0, column: 4 },
          { row: 0, column: 5 }
        ]
      };
    case 2:
      return {
        value: 3,
        turn: 0,
        coords: [
          { row: -2, column: 5 },
          { row: -1, column: 5 },
          { row: 0, column: 4 },
          { row: 0, column: 5 }
        ]
      };
    case 3:
      return {
        value: 4,
        turn: 0,
        coords: [
          { row: -1, column: 4 },
          { row: -1, column: 5 },
          { row: 0, column: 5 },
          { row: 0, column: 6 }
        ]
      };
    case 4:
      return {
        value: 5,
        turn: 0,
        coords: [
          { row: -1, column: 5 },
          { row: -1, column: 6 },
          { row: 0, column: 4 },
          { row: 0, column: 5 }
        ]
      };
    case 5:
      return {
        value: 6,
        turn: 0,
        coords: [
          { row: -3, column: 4 },
          { row: -2, column: 4 },
          { row: -1, column: 4 },
          { row: 0, column: 4 }
        ]
      };
    case 6:
      return {
        value: 7,
        turn: 0,
        coords: [
          { row: -2, column: 4 },
          { row: -1, column: 4 },
          { row: -1, column: 5 },
          { row: 0, column: 4 }
        ]
      };
    default:
      return {
        value: 1,
        turn: 0,
        coords: [
          { row: 0, column: 4 },
          { row: -1, column: 4 },
          { row: 0, column: 5 },
          { row: -1, column: 5 }
        ]
      };
  }
}
