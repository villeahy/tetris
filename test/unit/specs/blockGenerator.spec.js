import blockGenerator from '@/components/tetris/blockGenerator'

test('works', () => {
  expect(blockGenerator(0)).toMatchSnapshot()
  expect(blockGenerator(1)).toMatchSnapshot()
  expect(blockGenerator(2)).toMatchSnapshot()
  expect(blockGenerator(3)).toMatchSnapshot()
  expect(blockGenerator(4)).toMatchSnapshot()
  expect(blockGenerator(5)).toMatchSnapshot()
  expect(blockGenerator(6)).toMatchSnapshot()
  expect(blockGenerator('asd')).toMatchSnapshot()
})
