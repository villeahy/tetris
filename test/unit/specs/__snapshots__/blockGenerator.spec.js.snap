// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
Array [
  Object {
    "i": 0,
    "y": 4,
  },
  Object {
    "i": -1,
    "y": 4,
  },
  Object {
    "i": 0,
    "y": 5,
  },
  Object {
    "i": -1,
    "y": 5,
  },
]
`;

exports[`works 2`] = `
Array [
  Object {
    "i": -2,
    "y": 4,
  },
  Object {
    "i": -1,
    "y": 4,
  },
  Object {
    "i": 0,
    "y": 4,
  },
  Object {
    "i": 0,
    "y": 5,
  },
]
`;

exports[`works 3`] = `
Array [
  Object {
    "i": -2,
    "y": 5,
  },
  Object {
    "i": -1,
    "y": 5,
  },
  Object {
    "i": 0,
    "y": 4,
  },
  Object {
    "i": 0,
    "y": 5,
  },
]
`;

exports[`works 4`] = `
Array [
  Object {
    "i": -1,
    "y": 4,
  },
  Object {
    "i": -1,
    "y": 5,
  },
  Object {
    "i": 0,
    "y": 5,
  },
  Object {
    "i": 0,
    "y": 6,
  },
]
`;

exports[`works 5`] = `
Array [
  Object {
    "i": -1,
    "y": 5,
  },
  Object {
    "i": -1,
    "y": 6,
  },
  Object {
    "i": 0,
    "y": 4,
  },
  Object {
    "i": 0,
    "y": 5,
  },
]
`;

exports[`works 6`] = `
Array [
  Object {
    "i": -3,
    "y": 4,
  },
  Object {
    "i": -2,
    "y": 4,
  },
  Object {
    "i": -1,
    "y": 4,
  },
  Object {
    "i": 0,
    "y": 4,
  },
]
`;

exports[`works 7`] = `
Array [
  Object {
    "i": -2,
    "y": 4,
  },
  Object {
    "i": -1,
    "y": 4,
  },
  Object {
    "i": -1,
    "y": 5,
  },
  Object {
    "i": 0,
    "y": 4,
  },
]
`;

exports[`works 8`] = `
Array [
  Object {
    "i": 0,
    "y": 4,
  },
  Object {
    "i": -1,
    "y": 4,
  },
  Object {
    "i": 0,
    "y": 5,
  },
  Object {
    "i": -1,
    "y": 5,
  },
]
`;
